File: Fase Zero SBC/f.py
n = int(input())
m = int(input())

dna = str(input())
pro = str(input())

casos = int(input())

for i in range(casos):
    a = int(input())
    b = int(input())

    print(dna.count(pro[a - 1:b]))



# aaa

File: Fase Zero SBC/f.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  string dna;
  cin >> dna;

  string pro;
  cin >> pro;

  int casos;
  cin >> casos;

  for (int i = 0; i < casos; i++) {
    int a, b;
    cin >> a >> b;

    string substring = pro.substr(a - 1, b);
    int count = 0;
    size_t index = dna.find(substring);
    while (index != string::npos) {
      count++;
      index = dna.find(substring, index + 1);
    }
    cout << count << endl;
  }

  return 0;
}